src/EventStore.Core/Services/SubscriptionMetric.cs

```diff
@@ -0,0 +1,74 @@
+#nullable enable
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using EventStore.Core.Messages;
+
+namespace EventStore.Core.Services;
+
+public class SubscriptionMetric {
+	private readonly ConcurrentDictionary<Guid, MonitoringMessage.SubscriptionStats> _subscriptionStats;
+
+	public SubscriptionMetric(Meter meter, string name) {
+		_subscriptionStats = new();
+
+		meter.CreateObservableUpDownCounter($"{name}-completed", Completed);
+		meter.CreateObservableUpDownCounter($"{name}-subscription-position", SubscriptionPosition);
+		meter.CreateObservableUpDownCounter($"{name}-stream-position", StreamPosition);
```

Comment:
i don't think there is an easy way to tell how many subscriptions there currently are (?) if not, it might be worth adding that as a separate metric